core/Azure.Mcp.Core/src/Services/Azure/Tenant/TenantService.cs

```diff
@@ -1,20 +1,32 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Azure.Core;
+using Azure.Mcp.Core.Services.Azure.Authentication;
 using Azure.Mcp.Core.Services.Caching;
 using Azure.ResourceManager;
 using Azure.ResourceManager.Resources;
 
 namespace Azure.Mcp.Core.Services.Azure.Tenant;
 
-public class TenantService(ICacheService cacheService)
-    : BaseAzureService, ITenantService
+public class TenantService: BaseAzureService, ITenantService
```

Comment:
With `TenantService` now responsible for providing credentials, I’m wondering if there’s still a need for it to inherit from `BaseAzureService`?, if not it will let us avoid exposing the protected `TenantService` property in `BaseAzureService`, which is a workaround to break the circular dependency, thinking loud

Reply:
Totally agreed. However, that will touch a lot of stuff that I'd rather do in a follow up PR. I want to have such a dramatic change separate from the functionality added in this PR.